autopush-common/src/db/bigtable/bigtable_client/mod.rs

```diff
@@ -836,37 +835,28 @@ impl DbClient for BigTableClientImpl {
     async fn get_channels(&self, uaid: &Uuid) -> DbResult<HashSet<Uuid>> {
         let mut result = HashSet::new();
 
-        let req = {
-            let filter = {
-                let mut strip_filter = data::RowFilter::default();
-                strip_filter.set_strip_value_transformer(true);
-                let mut regex_filter = data::RowFilter::default();
-                // Your regex expression must match the WHOLE string. No partial matches.
-                // For this, we only want to grab the channel meta records (which do not
-                // have a sort key suffix)
-                let key = format!("^{}#[^#]+", uaid.simple());
-                regex_filter.set_row_key_regex_filter(key.as_bytes().to_vec());
-                let mut chain = data::RowFilter_Chain::default();
-                let mut repeat_field = RepeatedField::default();
-                repeat_field.push(strip_filter);
-                repeat_field.push(regex_filter);
-                chain.set_filters(repeat_field);
-
-                let mut filter = data::RowFilter::default();
-                filter.set_chain(chain);
-                filter
-            };
-
-            let mut req = bigtable::ReadRowsRequest::default();
-            req.set_table_name(self.settings.table_name.clone());
-            req.set_filter(filter);
+        let mut req = bigtable::ReadRowsRequest::default();
+        req.set_table_name(self.settings.table_name.clone());
 
-            req
-        };
+        let start_key = format!("{}@", uaid.simple());
+        // '[' is the char after '@'
+        let end_key = format!("{}[", uaid.simple());
+        let mut rows = data::RowSet::default();
+        let mut row_range = data::RowRange::default();
+        row_range.set_start_key_open(start_key.into_bytes());
+        row_range.set_end_key_open(end_key.into_bytes());
```

Comment:
That's what `open` does -- it's the exclusive option whereas closed is the inclusive. So this would exclude anything beginning with `...@[`.

However I do use a closed in `fetch_timestamp_messages` for `..#02:` which you could make the same case for, so I've changed that to open.